packages/react-sdk/src/RiverContext.tsx

```diff
@@ -0,0 +1,25 @@
+'use client'
+import { SignerContext, SyncAgent } from '@river-build/sdk'
+import { createContext } from 'react'
+
+type RiverContext = {
+    signerContext: SignerContext
+    syncAgent: SyncAgent
+}
+
+export const RiverContext = createContext<RiverContext | undefined>(undefined)
+
+type RiverProviderProps = {
+    signerContext: SignerContext
+    syncAgent: SyncAgent
+    children?: React.ReactNode
+}
```

Comment:
if we require a syncAgent here we don't need the signerContext too, the syncAgent contains the signerContext

Reply:
If that's the case, I think that we only need one context: `<RiverSyncContext syncAgent={syncAgent} />` 
Is there any other case that we would like to have a more granular approach?